File: frontend/src/types/account.ts
export type Broker = "robinhood" | "merrill" | "other";

export interface Account {
  id: string;
  user_id: string;
  name: string;
  broker: Broker;
  tax_treatment: string | null;
  created_at: string;
  updated_at: string;
}

export interface AccountCreate {
  name: string;
  broker: Broker;
  tax_treatment?: string | null;
}

export interface AccountUpdate {
  name?: string;
  broker?: Broker;
  tax_treatment?: string | null;
}

File: backend/app/schemas/account.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict

from app.schemas.enums import Broker


class AccountCreate(BaseModel):
    model_config = ConfigDict(strict=False)

    name: str
    broker: Broker
    tax_treatment: Optional[str] = None


class AccountUpdate(BaseModel):
    model_config = ConfigDict(strict=False)

    name: Optional[str] = None
    broker: Optional[Broker] = None
    tax_treatment: Optional[str] = None


class AccountResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    name: str
    broker: str
    tax_treatment: Optional[str]
    created_at: datetime
    updated_at: datetime
